runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java

```diff
@@ -0,0 +1,207 @@
+	private final ItemManager itemManager;
+	private final ScheduledExecutorService executor;
+
+	private ItemClient itemClient;
+
+	private IconTextField searchBox = new IconTextField();
+	private JPanel searchItemsPanel = new JPanel();
+	private JLabel searchingLabel = new JLabel("Searching...");
+
+	GrandExchangeSearchPanel(Client client, ItemManager itemManager, ScheduledExecutorService executor)
+	{
+		this.client = client;
+		this.itemManager = itemManager;
+		this.executor = executor;
+		init();
+	}
+
+	void init()
+	{
+		setLayout(new BorderLayout());
+
+		// Search Box
+		Icon search;
+		try
+		{
+			search = new ImageIcon(ImageIO.read(GrandExchangePlugin.class.getResourceAsStream("search.png")));
+		}
+		catch (IOException ex)
+		{
+			throw new RuntimeException(ex);
+		}
+
+		searchBox.setIcon(search);
+		searchBox.addActionListener(e -> executor.execute(this::priceLookup));
+
+		add(searchBox, BorderLayout.NORTH);
+
+		// Searching label
+		searchingLabel.setHorizontalAlignment(JLabel.CENTER);
+		searchingLabel.setForeground(Color.YELLOW);
+
+		// Items Panel
+		searchItemsPanel.setLayout(new GridLayout(0, 1, 0, 3));
+		searchItemsPanel.setBorder(new EmptyBorder(3, 0, 0, 0));
+
+		add(searchItemsPanel, BorderLayout.SOUTH);
+
+	}
+
+	void priceLookup(String item)
+	{
+		searchBox.setText(item);
+		executor.execute(this::priceLookup);
+	}
+
+	private void priceLookup()
+	{
+		String lookup = searchBox.getText();
+
+		if (Strings.isNullOrEmpty(lookup))
+		{
+			return;
+		}
+
+		// Input is not empty, add searching label
+		searchItemsPanel.removeAll();
+		add(searchingLabel, BorderLayout.CENTER);
+		revalidate();
+
+		SearchResult result;
+
+		try
+		{
+			result = itemManager.searchForItem(lookup);
+		}
+		catch (ExecutionException ex)
+		{
+			log.warn("Unable to search for item {}", lookup, ex);
+			removeSearch();
+			return;
+		}
+
+
+		if (result != null && !result.getItems().isEmpty())
+		{
+			itemClient = new ItemClient();
+
+			for (Item item : result.getItems())
+			{
+				int itemId = item.getId();
+
+				ItemComposition itemComp = client.getItemDefinition(itemId);
+				if (itemComp == null)
+				{
+					continue;
+				}
+
+				ItemPrice itemPrice;
+				try
+				{
+					itemPrice = itemManager.getItemPrice(itemId);
+				}
+				catch (IOException ex)
+				{
+					log.warn("Unable to fetch item price for {}", itemId, ex);
+					removeSearch();
+					return;
+				}
+
+				BufferedImage itemImage;
+				try
+				{
+					itemImage = itemClient.getIcon(itemId);
+				}
+				catch (IOException ex)
+				{
+					log.warn("Unable to fetch item icon for {}", itemId, ex);
+					removeSearch();
+					return;
+				}
+
+				if (itemImage == null)
+				{
+					log.warn("Unable to fetch item icon for {}", itemId);
+					removeSearch();
+					return;
+				}
+
+				SwingUtilities.invokeLater(() ->
+				{
+					searchItemsPanel.add(new GrandExchangeItemPanel(itemImage, item.getName(), itemId, itemPrice.getPrice(), itemComp.getPrice() * 0.6));
```

Comment:
Maybe sorting these would be a good idea. But at least start with adding each of the items to some local list and then call SwingUtilities.invokeLater() and iterate through the list to reduce the calls to invokeLater